feat: LEAP-58: Proxy URL for Taxonomy if possible

robot-ci-heartex opened the pull request:
Hi @hlomzik!

This PR was [created](https://github.com/HumanSignal/label-studio/actions/runs/6553296978) in a response to PRs in upstream repos:
- https://github.com/HumanSignal/dm2/pull/251
- https://github.com/HumanSignal/label-studio-frontend/pull/1592

hlomzik committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6553296978

hlomzik committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6553340386

hlomzik committed:
[submodules] Build static HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6578959553

hlomzik committed:
Merge branch 'origin/develop' into 'fb-leap-58/proxy-url'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6578959553

jombooth committed:
[submodules] Build static HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6619537357

jombooth committed:
Merge branch 'origin/develop' into 'fb-leap-58/proxy-url'

jombooth committed:
fix: LEAP-58: Add project-level presign API to support flat-file taxonomy in storage

jombooth committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6633835281

jombooth committed:
Merge branch 'origin/develop' into 'fb-leap-58/proxy-url'

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6633857155

jombooth committed:
fix paths issue

jombooth committed:
fix failing pytests

jombooth committed:
associated frontend updates

jombooth committed:
merge develop and rebuild frontend

jombooth committed:
introduce some tavern tests for new + existing endpoint

jombooth committed:
merge in second branch

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6660611165

jombooth committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6661908415

jombooth committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6661911849

jombooth committed:
additional function for freezing time

jombooth committed:
Merge branch 'fb-leap-58/proxy-url' of github.com:heartexlabs/label-studio into fb-leap-58/proxy-url

dredivaris commented on label_studio/data_import/api.py:
```diff
@@ -712,3 +696,51 @@ def get(self, request, *args, **kwargs):
+        fileuri = request.GET.get('fileuri')
+
+        if fileuri is None or task_id is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```
Not sure 404 is the right status code for a request with missing data.  A more appropriate response code would be 400 bad request (generally due to client error - in this case missing params)

dredivaris commented on label_studio/data_import/api.py:
```diff
@@ -712,3 +696,51 @@ def get(self, request, *args, **kwargs):
+        fileuri = request.GET.get('fileuri')
+
+        if fileuri is None or project_id is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```
Same issue as above comment

dredivaris reviewed (dismissed):
LGTM except for 2 response code change suggestions

MihajloHoma committed:
[submodules] Build static HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6717346168

MihajloHoma committed:
docs: LSF Update

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6717346168

jombooth committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6725009387

jombooth committed:
merge develop / accept all incoming dist changes in preparation for rebuild

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6725215270

jombooth committed:
404 -> 400

jombooth committed:
Merge branch 'fb-leap-58/proxy-url' of github.com:heartexlabs/label-studio into fb-leap-58/proxy-url

jombooth committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6725281641

jombooth committed:
fix testcases

jombooth committed:
Merge branch 'fb-leap-58/proxy-url' of github.com:heartexlabs/label-studio into fb-leap-58/proxy-url

hlomzik committed:
[submodules] Build static HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6725543628

hlomzik committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6725633362

robot-ci-heartex commented:
To enable Auto Merge for this PR also merge those PRs:
- https://github.com/HumanSignal/label-studio-frontend/pull/1592

jombooth committed:
fix comment

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6725767581

hlomzik committed:
[submodules] Build static HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6725940200
